New warning from boost

I now get this:
```
[92/1227] Building CXX object bundled/boost-1.62.0/libs/system/src/CMakeFiles/obj_boost_system.debug.dir/error_code.cpp.o
In file included from /u/bangerth/p/deal.II/2/dealii/bundled/boost-1.62.0/libs/system/src/error_code.cpp:16:0:
/u/bangerth/p/deal.II/2/dealii/bundled/boost-1.62.0/include/boost/system/error_code.hpp:221:36: warning: ‘boost::system::posix_category’ defined but not used [-Wunused-variable]
     static const error_category &  posix_category = generic_category();
                                    ^
/u/bangerth/p/deal.II/2/dealii/bundled/boost-1.62.0/include/boost/system/error_code.hpp:222:36: warning: ‘boost::system::errno_ecat’ defined but not used [-Wunused-variable]
     static const error_category &  errno_ecat     = generic_category();
                                    ^
/u/bangerth/p/deal.II/2/dealii/bundled/boost-1.62.0/include/boost/system/error_code.hpp:223:36: warning: ‘boost::system::native_ecat’ defined but not used [-Wunused-variable]
     static const error_category &  native_ecat    = system_category();
                                    ^
```
These are global variables that appear to be unused. The warning is only shown once when we build the one boost-lib file.

Can we mark them with `DEAL_II_UNUSED` to make the warning go away?

--- a/bundled/boost-1.62.0/include/boost/system/error_code.hpp
+++ b/bundled/boost-1.62.0/include/boost/system/error_code.hpp
@@ -218,9 +218,9 @@ namespace boost
     inline const error_category &  get_system_category() { return system_category(); }
     inline const error_category &  get_generic_category() { return generic_category(); }
     inline const error_category &  get_posix_category() { return generic_category(); }
-    static const error_category &  posix_category = generic_category();
-    static const error_category &  errno_ecat     = generic_category();
-    static const error_category &  native_ecat    = system_category();
+    static const error_category &  posix_category BOOST_ATTRIBUTE_UNUSED = generic_category();
+    static const error_category &  errno_ecat     BOOST_ATTRIBUTE_UNUSED = generic_category();
+    static const error_category &  native_ecat    BOOST_ATTRIBUTE_UNUSED = system_category();
 # endif
 
     //  class error_condition  -----------------------------------------------//

